esp/esp/program/modules/handlers/teacherbigboardmodule.py

```diff
@@ -0,0 +1,206 @@
+    # time the page refreshes for the same admin, they will get new numbers
+
+    @cache_function_for(105)
+    def num_teachers_teaching(self, prog):
+        # Querying for SRs and then extracting the users saves us joining the
+        # users table.
+        return ClassSubject.objects.filter(parent_program=prog
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_active_users(self, prog, minutes=10):
+        recent = datetime.datetime.now() - datetime.timedelta(0, minutes * 60)
+        return ClassSubject.objects.filter(parent_program=prog, timestamp__gt=recent
+        ).values_list('teachers').distinct().count()
+
+    @cache_function_for(105)
+    def num_checked_in_teachers(self, prog):
+        return Record.objects.filter(program=prog, event='teacher_checked_in').count()
+
+    @cache_function_for(105)
+    def num_class_reg(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog).values_list('id').count()
+
+    @cache_function_for(105)
+    def num_class_app(self, prog):
+        return ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0).count()
+
+    @cache_function_for(105)
+    def reg_classes(self, prog):
+        class_times = dict(ClassSubject.objects.filter(parent_program=prog
+        ).values_list('id').annotate(Min('timestamp')))
+        return sorted(class_times.itervalues())
+
+    @cache_function_for(105)
+    def app_classes(self, prog):
+        #all ClassSubjects that are approved (and have an approved section)
+        class_times = dict(ClassSubject.objects.filter(parent_program=prog, status__gt=0, sections__status__gt=0
+        ).values_list('id').annotate(Min('timestamp')))
+        return sorted(class_times.itervalues())
+
+    @cache_function_for(105)
+    def teach_times(self, prog):
+        teacher_times = dict(ClassSubject.objects.filter(parent_program=prog
+        ).values_list('teachers').annotate(Min('timestamp')))
+        return sorted(teacher_times.itervalues())
+
+    @cache_function_for(105)
+    def get_hours(self, prog):
+        hours = ClassSubject.objects.filter(parent_program=prog
+        ).exclude(category__category__iexact="Lunch").values_list('id','duration','class_size_max'
+        ).annotate(Min('timestamp'))
+        sorted_hours = sorted(hours, key=operator.itemgetter(2))
+        return [hour[1:] for hour in sorted_hours]
+
+    @cache_function_for(105)
+    def static_hours(self, prog):
+        hours = ClassSubject.objects.filter(parent_program=prog
+        ).exclude(category__category__iexact="Lunch").values_list('id','duration','class_size_max'
+        )
+        hours = [[float(j[1])] + [float(j[1])*j[2]] for j in hours]
+        return [sum(j) for j in zip(*hours)]
+
+    @staticmethod
+    def make_hours_data(hourss, drop_beg = 0, drop_end = 0, cutoff = 0, delta=datetime.timedelta(0, 3600)):
```

Comment:
Hmm, why do you need a whole separate method here?  Can't you just pass a list of durations and a list of `class_size_max * duration`s to `make_graph_data`?  (In fact it seems like you already do much of that work in `static_hours`!  You could just return a line earlier and use the data one way for the graphs and the other way for the numbers.

Reply:
Ah, I see.  yeah, I'd say do exactly that.